### DISCUSSION TITLE
Suggestion: per queue message size limit

### DISCUSSION THREAD

**Discussion by johanrhodin** [rabbitmq/rabbitmq-server#10907] [Ideas]
An operator can limit the global message size with `max_message_size`. For some environments this is too broad. A user might want to limit the message size in more detail for only some queues (such as quorum queues), but not other queues (classic queues, streams), or for queues `A`, `B`, `C`, but not `D`. This could be designed as a policy.

Before doing a full implementation we (CloudAMQP) wanted to check if there are any opinions on this in the community.

**Comment by michaelklishin** [comment]
I don't remember seeing similar requests.

`max_message_size` is enforced by channels (`rabbit_channel:check_msg_size/2`) and AMQP 1.0 sessions (`rabbit_amqp_session:handle_control/1`), neither of which is queue (at the time of limit enforcement) or policy aware.

Enforcing limits at the queue level means risky changes to the queue <=> channel communication "protocol" and to me, the benefit does not seem to be worth that risk.

**Comment by michaelklishin** [comment]
One variation of this that would be easy to implement with no risks that I can think of is a virtual host-level limit. Channels and AMQP 1.0 sessions can fetch it when they initialize.
